### PR TITLE
Added workaround for FIELD function

### PR DESCRIPTION
**Pull request by deeptikorwar**
Addresses #4962

@rmloveland, I've added the workaround details in 20.1, but looks like the syntax is not quite right. 

I tried the example in 20.1  and although both `SELECT array_position(ARRAY[4,1,3,2],1);` and  `SELECT * FROM table_a ORDER BY 2` work, `SELECT * FROM table_a ORDER BY array_position(ARRAY[4,1,3,2],1);` doesn't.

### CODE REVIEW

File: v20.1/migrate-from-mysql.md [hunk 1]
```diff
@@ -20,6 +20,31 @@ MySQL strings are case-insensitive by default, but strings in CockroachDB are ca
 
 For more information about the case sensitivity of strings in MySQL, see [Case Sensitivity in String Searches](https://dev.mysql.com/doc/refman/8.0/en/case-sensitivity.html) from the MySQL documentation.  For more information about CockroachDB strings, see [`STRING`](string.html).
 
+### `FIELD` function
+
+The MYSQL `FIELD` function is not supported in CockroachDB. Instead, you can use the [`array_position`](functions-and-operators.html#array-functions) function, which returns the index of the first occurrence of element in the array.
+
+While MYSQL returns 0 when the element is not found, CockroachDB returns `NULL`. So if you are using the `ORDER BY` clause in a statement with the `array_position` function, the caveat is that sort is applied even when the element is not found. As a workaround, you can use the [`COALESCE`](functions-and-operators.html#conditional-and-function-like-operators) operator.
```
Comment (rmloveland) [inline v20.1/migrate-from-mysql.md:27]: Can you please move this paragraph above the `COALESCE` example? (I have added another comment with 'X' to mark the spot)

That way the reader can read in a potentially nicer order:

1) intro
2) simple example
3) this paragraph with the more involved description of the workaround
4) the workaround

File: v20.1/migrate-from-mysql.md [hunk 2]
```diff
@@ -20,6 +20,31 @@ MySQL strings are case-insensitive by default, but strings in CockroachDB are ca
...
+> SELECT array_position(ARRAY[4,1,3,2],1);
+~~~
+
+~~~
+  array_position
+------------------
+               2
+(1 row)
+~~~
+
```
Comment (rmloveland) [inline v20.1/migrate-from-mysql.md:42]: X

**Review by rmloveland — CHANGES_REQUESTED** [review]
This is really great @deeptikorwar !  Thank you for submitting this! I have requested only one change in the ordering, but otherwise it's very nice.

PS Sorry for the slow response on my end, we are in an especially busy time getting ready for the 20.2 release.

File: v20.1/migrate-from-mysql.md (hunk 1 above)
Comment (deeptikorwar) [reply v20.1/migrate-from-mysql.md]: Thank you.

**Comment by rmloveland** [comment]
Hi @deeptikorwar, can you please change this PR so it only adds this info to the following versions:

- 19.2
- 20.1
- 20.2


We are releasing 20.2 very soon (this month), and our support policy (IIRC) is "current version plus last 2 releases".

Thanks!  This is really great, as soon as you change to affect just these versions, we can merge this immediately.

**Comment by deeptikorwar** [comment]
Thanks @rmloveland. I've taken those files out.

**Review by rmloveland — APPROVED** [review]

**Comment by rmloveland** [comment]
Thanks @deeptikorwar !  Very nice contribution.  Merging now :-)

**Pull request merged**